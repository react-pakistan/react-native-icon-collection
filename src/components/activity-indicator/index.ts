export {
  ActivityIndicator,
  ActivityIndicatorSize,
} from './activity-indicator';
