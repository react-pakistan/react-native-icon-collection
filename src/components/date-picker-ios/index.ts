export { DatePickerIOS } from './date-picker-ios';
