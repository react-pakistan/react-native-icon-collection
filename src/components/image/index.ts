export {
  IImageProps,
  Image,
  ResizeMethod,
  ResizeMode,
} from './image';
