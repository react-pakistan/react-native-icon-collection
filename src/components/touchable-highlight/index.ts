export { TouchableHighlight } from './touchable-highlight';
