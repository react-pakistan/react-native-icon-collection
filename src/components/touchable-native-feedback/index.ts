export { TouchableNativeFeedback } from './touchable-native-feedback';
