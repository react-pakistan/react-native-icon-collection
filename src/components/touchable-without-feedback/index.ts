export { TouchableWithoutFeedback } from './touchable-without-feedback';
