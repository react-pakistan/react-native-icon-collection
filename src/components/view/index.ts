export { View, IView } from './view';
