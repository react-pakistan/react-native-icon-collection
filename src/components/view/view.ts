import { ITheme } from '@react-pakistan/react-commons-collection/theme';
import { ThemedStyledProps } from 'styled-components';
import styled from 'styled-components/native';

export const View = styled.View`
  align-items: ${({ alignItems } : IView) : string => alignItems || 'center'};
  background-color: ${({ backgroundColor, theme } : ThemedStyledProps<IView, ITheme>) : string => backgroundColor || theme.colors.cherryRed};
  flex-direction: ${({ flexDirection } : IView) : string => flexDirection || 'column'};
  flex: ${({ flex } : IView) : number => flex || 1};
  justify-content: ${({ justifyContent } : IView) : string => justifyContent || 'center'};
`;

export interface IView {
  /**
   * Specifies alignItems for View
   * @default center
   */
  alignItems? : string;
  /**
   * Specifies backgroundColor for View
   * @default theme.color.jcWhite;
   */
  backgroundColor? : string;
  /**
   * Specifies flexDirection for View
   * @default column
   */
  flexDirection? : string;
  /**
   * Specifies flex for View
   * @default 1
   */
  flex? : number;
  /**
   * Specifies justifyContent for View
   * @default center
   */
  justifyContent? : string;
}
