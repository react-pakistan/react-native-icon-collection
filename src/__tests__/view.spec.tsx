import React from 'react';
import renderer from 'react-test-renderer';
import { View } from '../components/view';
import {
  defaultColumnViewProps,
  defaultRowViewProps,
} from '../../__stories__/story/view.story';

describe('Testing A component', () => {
  it('matches A snapshot, flex column', () => {
    const wrapper = renderer.create(
      <View {...defaultColumnViewProps} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('matches A snapshot, flex row', () => {
    const wrapper = renderer.create(
      <View {...defaultRowViewProps} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
