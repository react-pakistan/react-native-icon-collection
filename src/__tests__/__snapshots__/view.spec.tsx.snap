// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing A component matches A snapshot, flex column 1`] = `
<View
  alignItems="center"
  backgroundColor="white"
  flex={1}
  flexDirection="column"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`Testing A component matches A snapshot, flex row 1`] = `
<View
  alignItems="center"
  backgroundColor="white"
  flex={1}
  flexDirection="row"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;
